### PR TITLE
Refactor encrypted password reset so logic isn't duplicated

### PR DESCRIPTION
**Pull request by ageis**
@garrettr thanks for your review.

### CODE REVIEW

File: lib/encrypt_mails.rb [hunk 1]
```diff
@@ -151,51 +150,12 @@ def reset(header)
...
-          recipient[:unchanged][:to].push user if user_key.nil?
-        else
-          recipient[:lost][:to].push user
-      end
-
-      recipient
-
+      users = Array.new
+      users.push user
+      headers[:to] = users
```
Comment (garrettr) [inline lib/encrypt_mails.rb]: This whole function could be one line: `headers[:to] = [User.find_by_mail(headers[:to])]`.

There should also be a comment explaining why this is being done.

File: lib/encrypt_mails.rb [hunk 2]
```diff
@@ -32,15 +32,14 @@ def mail_with_relocation(headers={}, &block)
       act = Setting.plugin_openpgp['activation']
       return mail_without_relocation(headers, &block) if
         act == 'none' or not actions.include? @_action_name or
-        (act == 'project' and not project.try('module_enabled?', 'openpgp'))
+        (act == 'project' and not project.try('module_enabled?', 'openpgp') and not @_action_name == 'lost_password')
```
Comment (garrettr) [inline lib/encrypt_mails.rb:36]: Why is this necessary? I am confused. Shouldn't `actions.include? @_action_name` return True for `lost_password`?

File: lib/encrypt_mails.rb (hunk 1 above)
Comment (garrettr) [inline lib/encrypt_mails.rb]: You might need to add a line of `headers` so it gets returned. No need for `return`.

**Comment by garrettr** [comment]
lgtm :ship:

**Comment by ageis** [comment]
In programming there's always many approaches to solving a problem. You were right, as usual!
